simulator/func_sim/rf/rf.h

```diff
@@ -79,13 +84,29 @@ class RF
         else if ( accumulating_instr == -1)
             val = read_hi_lo() - val;
 
+        if( loadlr)
+        {
+            if(loadlr > 0)
+            {
+                uint32 reg_val = static_cast<uint32>(get_value( num)); // read register value
+                int8 shift = static_cast<int8>(val >> 24);             // find out the number of bytes that I should load
+                reg_val &= mask[shift - 1];                            // delete right bytes and save the rest
```

Comment:
You commented obvious things but left the hardest ones:
1. Why shift is taken from (val >> 24)? 
2. What does -1 and +2?